### PR TITLE
Ansel

### LINKED ISSUE #35: Multiple Issues with Saving, Opening, and Copy URL
> Ensure saving and opening functionality supports all new features. Similar to the URL, saving currently does not encode all the information needed to restore the entire linkage state. It is essential to have a seamless save and open experience to prevent user frustration and loss of work. By updating this feature, users can be confident that their work will be preserved and easily accessible when needed.

Your task is to make the 'Share Project' button the toolbar functional again. When the user pushes that button they could get a URL in their clipboard. When they enter this URL into another browser, the entire project state should be recovered from the data in the URL. The URL should be as short as reasonably possible while still supporting up to around 100 joints, links, and forces total.

**Specifically, the URL needs to contain the following info:**
A list of all joints (id, x, y, isGrounded, isInput)
A list of all links (id, joints contained in link, forces contained in link, mass, MOI, sublinks, color)
A list of all forces (id, startCoord, endCoord, isLocal, isFacingOut, magnitude)
A hashmap that maps ids to custom names (in custom-id service)
All settings (Global Units, Angle Units, input speed, input direction, is gravity on, is grid on, is minorGirdLines on?, object scale, currentTimestep from the animation bar)

**These functions will be most relevant:**
src/app/services/url-processor.service.ts ->  constructor()
 src/app/component/toolbar/toolbar.component.ts -> copyURL()

**Other relevant files:**
src/app/model/joint.ts
src/app/model/link.ts
src/app/model/force.ts
src/app/component/toolbar/toolbar.component.html
src/app/component/settings-panel/settings-panel.component.ts
src/app/services/settings.service.ts
src/app/services/custom-id.service.ts

Copy URL does not work and we might need to improve the save file format. For the URL, Ansel has already shortened it significantly but need more testing

### PR DESCRIPTION
**Pull request by AnselChang**
Resolves #35

### CODE REVIEW

**Comment by Kohmei358** [comment]
Sliders don't look like they are saving:
Screenshot 2023-06-06 at 1 41 07 PM
Screenshot 2023-06-06 at 1 40 59 PM

Also, the force arrow seems visually broken:
Screenshot 2023-06-06 at 1 41 07 PM
Screenshot 2023-06-06 at 1 40 59 PM

**Comment by AnselChang** [comment]
Rip. Will take a look.

File: count.sh
Comment (Kohmei358) [inline count.sh:1]: Did you mean to push this?

File: src/app/component/toolbar/toolbar.component.ts
Comment (Kohmei358) [inline src/app/component/toolbar/toolbar.component.ts:1]: Would you want to move the copy URL logic into the URL Processor service (or a new service)? I feel like that will further separate the app logic from the frontend.

File: src/app/services/grid-utils.service.ts
Comment (Kohmei358) [inline src/app/services/grid-utils.service.ts:1]: Accidental?

**Comment by AnselChang** [comment]
Fixed sliders

**Comment by Kohmei358** [comment]
Fixes #93 too

**Review by Kohmei358 — APPROVED** [review]
Looks good! I merged my templates UI. I'm working on a better design but can you check the functionality of everything and merge?

**Comment by AlexG1031** [comment]
Played around with the URLs and they appear to be working properly! Can't wait to begin using them!

**Comment by AnselChang** [comment]
Awesome!

**Pull request merged**